rtos/Thread.cpp

```diff
@@ -23,6 +23,15 @@
 
 #include "mbed.h"
 #include "rtos/rtos_idle.h"
+#include "mbed_assert.h"
+
+#define ALIGN_UP(pos, align) ((pos) % (align) ? (pos) +  ((align) - (pos) % (align)) : (pos))
```

Comment:
We don't need to complicate it with conditionals, how about standard: `(pos + align - 1) & ~(align - 1)`

Reply:
Ok fair enough.